backend/storage-service/src/jvmMain/kotlin/file/services/acl/AclService.kt

```diff
@@ -287,6 +296,110 @@ class AclService(
             .toMap()
     }
 
+    private suspend fun updateSynchronizationType(path: String) {
+        data class SynchronizedFolderWithNewAccess(
+            val id: String,
+            val user: String,
+            val path: String,
+            val oldType: SynchronizationType,
+            val newType: SynchronizationType?
+        )
+
+        db.withSession { session ->
+            val syncedChildren = session.sendPreparedStatement(
+                {
+                    setParameter("path", path)
+                },
+                """
+                    select id, user_id, path, access_type from storage.synchronized_folders
+                    where path like :path || '%'
```

Comment:
This will cause problems. This will end up catching `/home/foo/bar` and `/home/foobar/baz` if the path is `/home/foo`.

Reply:
You also need to make sure that you are only storing the _normalized_ version of paths in the `synchronize_folders` table. You might already be doing this but you should check.